src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -1387,35 +1390,12 @@ internal double GetPortVerticalOffset(PortModel portModel)
         ///     Reads inputs list and adds ports for each input.
         /// </summary>
         [Obsolete("RegisterInputPorts is deprecated, please use the InPortNamesAttribute, InPortDescriptionsAttribute, and InPortTypesAttribute instead.")]
-        public void RegisterInputPorts()
+        public void RegisterInputPorts(IEnumerable<PortData> portDatas)
```

Comment:
API Change: `RegisterInputPorts()` is now `RegisterInputPorts(IEnumerable<PortData> portDatas)`